File: PROGRAMMERS/네트워크.java
import java.util.*;
​
class Solution {
    
    static int answer = 0;
    static boolean[] visited;
    static Queue<Integer> q = new LinkedList<>();
    static boolean[] visited;
    
    public int solution(int n, int[][] computers) {
        visited = new boolean[n];
        
        for(int i = 0; i < n; i++) {
            if(visited[i] == false) {
                bfs(i, n, computers);
                answer++;
            }
        }
        
        return answer;
    }
    
    public void bfs(int now, int n, int[][] computers) {
        q.add(now);
        visited[now] = true;
        
        while(!q.isEmpty()) {
            int curr = q.remove();
            
            for(int i = 0; i < n; i++) {
                if(visited[i] == false && computers[curr][i] == 1) {
                    visited[i] = true;
                    q.add(i);
                }
            }
        }
    }
}

File: PROGRAMMERS/네트워크.cpp
#include <bits/stdc++.h>
using namespace std;

int N, answer;
bool visited[201];
vector<vector<int>> v;

void dfs(int start) {
    if(visited[start]) {
        return;
    }

    visited[start] = 1;

    for(int i = 0; i < v[start].size(); i++) {
        if(i == start || v[start][i] == 0) continue;
        dfs(i);
    }
}

int solution(int n, vector<vector<int>> computers) {

    N = n;
    v = computers;

    for(int i = 0; i < n; i++) {
        if(!visited[i]) {
            dfs(i);
            answer++;
        }
    }

    return answer;
}